commit/plugin.go

```diff
@@ -29,8 +41,10 @@ type Plugin struct {
 	lggr              logger.Logger
 	homeChain         reader.HomeChain
 	reportingCfg      ocr3types.ReportingPluginConfig
-	chainSupport      ChainSupport
-	observer          Observer
+	chainSupport      shared.ChainSupport
+	merkleWorker      merkleroot.Processor
+	tokenWorker       token.Processor
+	gasWorker         fees.Processor
```

Comment:
Do we want to have the concrete types here or the interfaces? I thought the interfaces would make for easier testing of the root plugin logic.